indico/modules/events/registration/controllers/__init__.py

```diff
@@ -40,11 +42,13 @@ def _process_POST(self):
         if self.management:
             session['registration_notify_user_default'] = notify_user
         modify_registration(self.registration, form_data, management=self.management, notify_user=notify_user)
-        return redirect(self.success_url)
+        return jsonify({'redirect': self.success_url})
 
     def _process_GET(self):
+        form_data = get_flat_section_submission_data(self.regform)
+        # Filter out registration data which are not in the regform anymore
```

Comment:
are you sure this is the old behavior? don't we still show those fields if they were used in the registration? i know for sure that we do this with removed options of choice fields...